src/blockchain/log-processors/order-created.ts

```diff
@@ -72,7 +75,58 @@ export function processOrderCreatedLog(db: Knex, augur: Augur, log: FormattedEve
 export function processOrderCreatedLogRemoval(db: Knex, augur: Augur, log: FormattedEventLog, callback: ErrorCallback): void {
   db.from("orders").where("orderId", log.orderId).delete().asCallback((err: Error|null): void => {
     if (err) return callback(err);
-    augurEmitter.emit("OrderCreated", log);
-    callback(null);
+    checkForUnOrphanedOrders(db, log, (err) => {
+      if (err) return callback(err);
+      augurEmitter.emit("OrderCreated", log);
+      return callback(null);
+    });
+  });
+}
+
+function checkForOrphanedOrders(db: Knex, orderData: OrdersRow<string>, callback: ErrorCallback): void {
+  const queryData = {
+    marketId: orderData.marketId,
+    outcome: orderData.outcome,
+    orderType: orderData.orderType,
+  };
+  db.first([db.raw("count(*) as numOrders"), db.raw("count(distinct(price)) as numPrices")]).from("orders").where(queryData).where("amount", ">", 0).asCallback((err: Error|null, results: { numOrders: number, numPrices: number }): void => {
```

Comment:
Does this comparison work since amount is a string?

Reply:
I think that makes more sense? Especially if we were to support other databases that, you know, care about types